google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1beta2/CivilTimeEncoderTest.java

```diff
@@ -0,0 +1,468 @@
+    // ->0x9C3F3F7EFB
+    assertEquals(LocalDateTime.of(9999, 12, 31, 23, 59, 59), CivilTimeEncoder.decodePacked64DatetimeSeconds(0x9C3F3F7EFBL));
+
+    //encode and decode failures
+    // 10000/01/01 00:00:00->0b0000000000000000000000|10011100010000|0001|00001|00000|000000|000000
+    // ->0x9C40420000
+    LocalDateTime dateTime = LocalDateTime.of(10000, 1, 1, 0, 0, 0);
+    try {
+      CivilTimeEncoder.encodePacked64DatetimeSeconds(dateTime);
+      Assert.fail();
+    } catch (IllegalArgumentException expected) {
+    }
+    try {
+      // 0001/01/01 00:00:00->0b0000000000000000000001|00000000000001|0001|00001|00000|000000|000000
+      // ->0x10004420000
+      CivilTimeEncoder.decodePacked64DatetimeSeconds(0x10004420000L);
+      Assert.fail();
+    } catch (IllegalArgumentException e) {
+    }
+    try {
+      // 0001/01/01 00:00:00->0b0000000000000000000001|00000000000001|0001|00001|00000|000000|000000
+      // ->0x10004420000
+      //invalid bit field
+      CivilTimeEncoder.decodePacked64DatetimeSeconds(0x10004420000L);
+      Assert.fail();
+    } catch (IllegalArgumentException e) {
+    }
+    try {
+      // 0001/01/01 00:00:60
+      // 0b0000000000000000000000|00000000000001|0001|00001|00000|000000|111100
+      // 0x442003C
+      CivilTimeEncoder.decodePacked64DatetimeSeconds(0x442003CL);
+      Assert.fail();
+    } catch (IllegalArgumentException e) {
+    }
+    try {
+      // 0001/01/01 00:60:00
+      // 0b0000000000000000000000|00000000000001|0001|00001|00000|111100|000000
+      // 0x4420F00
+      CivilTimeEncoder.decodePacked64DatetimeSeconds(0x4420F00L);
+      Assert.fail();
+    } catch (IllegalArgumentException e) {
+    }
+    try {
+      // 0001/01/01 24:00:00
+      // 0b0000000000000000000000|00000000000001|0001|00001|11000|000000|000000
+      // 0x4438000
+      CivilTimeEncoder.decodePacked64DatetimeSeconds(0x4438000L);
+      Assert.fail();
+    } catch (IllegalArgumentException e) {
+    }
+    try {
+      // 0001/01/00 00:00:00
+      // 0b0000000000000000000000|00000000000001|0001|00000|00000|000000|000000
+      // 0x4400000
+      CivilTimeEncoder.decodePacked64DatetimeSeconds(0x4400000L);
+      Assert.fail();
+    } catch (IllegalArgumentException e) {
+    }
+    try {
+      // 0001/13/01 00:00:00
+      // 0b0000000000000000000000|00000000000001|1101|00001|00000|000000|000000
+      // 0x7420000
+      CivilTimeEncoder.decodePacked64DatetimeSeconds(0x7420000L);
+      Assert.fail();
+    } catch (IllegalArgumentException e) {
+    }
+    try {
+      // 10000/01/01 00:00:00
+      // 0b0000000000000000000000|10011100010000|0001|00001|00000|000000|000000
+      // 0x9C40420000
+      CivilTimeEncoder.decodePacked64DatetimeSeconds(0x9C40420000L);
+      Assert.fail();
+    } catch (IllegalArgumentException e) {
+    }
+  }
+
+  @Test
+  public void Packed64DateTimeMicrosTest() {
+    //encode
```

Comment:
round trip testing?

Reply:
added

Reply:
I've added in the round trip for each set of functions.